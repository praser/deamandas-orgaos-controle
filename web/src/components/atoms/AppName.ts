import styled from 'styled-components';

const AppName = styled.h1`
  /* Layout */
  box-sizing: border-box;
  display: block;
  flex: 1;

  /* Text */
  font-family: Nunito, -apple-system, system-ui, 'Segoe UI', Roboto,
    'Helvetica Neue', Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
    'Segoe UI Symbol', 'Noto Color Emoji';
  font-size: 16px;
  font-weight: 800;
  letter-spacing: 0.8px;
  text-decoration-thickness: auto;
  text-size-adjust: 100%;
  text-transform: uppercase;

  /* Other */
  list-style-image: none;
  list-style-position: outside;
  list-style-type: none;
`;

export default AppName;
