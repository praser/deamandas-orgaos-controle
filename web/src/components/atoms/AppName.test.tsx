import React from 'react';
import { render, screen } from '@testing-library/react';
import AppName from './AppName';

describe('AppName atomic component', () => {
  const text = 'Nome do App';
  const testId = 'app-title';
  let appName: HTMLElement;

  beforeEach(() => {
    render(<AppName data-testid={testId}>{text}</AppName>);
    appName = screen.getByTestId(testId);
  });

  it('is expected to match snapshot', () => {
    expect(appName).toMatchSnapshot();
  });
});
