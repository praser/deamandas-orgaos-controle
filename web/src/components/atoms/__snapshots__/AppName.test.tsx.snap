// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppName atomic component is expected to match snapshot 1`] = `
<h1
  class="sc-bdvvaa jJMrBK"
  data-testid="app-title"
>
  Nome do App
</h1>
`;
